app/main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        Animal.alive.remove(self)
+
+    def damage(self, damage: int) -> None:
+        self.health -= damage
+        if self.health <= 0:
+            self.die()
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
```

Comment:
If this function only takes Herbivores, why would you need a check in the next line? Use parent class for annotations here to cover all cases